src/mount/readdata.cc

```diff
@@ -536,11 +579,13 @@ void read_data_init(uint32_t retries,
 	gChunkserverConnectTimeout_ms = chunkserverConnectTimeout_ms;
 	gChunkserverWaveReadTimeout_ms = chunkServerWaveReadTimeout_ms;
 	gChunkserverTotalReadTimeout_ms = chunkserverTotalReadTimeout_ms;
+	gOriginalCacheExpirationTime_ms = cache_expiration_time_ms;
 	gCacheExpirationTime_ms = cache_expiration_time_ms;
 	gReadaheadMaxWindowSize = readahead_max_window_size_kB * 1024;
 	gReadCacheMaxSize.store((read_chache_max_size_percentage * 0.01) *
 	                        gMemoryInfo->getTotalMemory());
 	gReadWorkers = read_workers;
+	kMaxWindowConsideringMaxReadCacheSize = gReadCacheMaxSize.load() / gReadWorkers;
```

Comment:
I usually use 256 read workers on powerfull servers and 64 MiB of max readahead window size. Does it means it will reduce the effective window size to 256 KiB?

Reply:
In that case that you are describing, you would need that the server you will use have at most 16 GiB. This is because in that case we should have `kMaxWindowConsideringMaxReadCacheSize <= 64 MiB`, and then in this [code](https://github.com/leil-io/saunafs/commit/09cf1ce8a0d7fb2cfd866eb86ee540d2c153b0a2#diff-7464618fee0a32f884909855b3a92331f9ba6649e3ab3fd5a2262b43b837f9e7L212), we should have `recommendedSize = kMaxWindowConsideringMaxReadCacheSize <= 64 MiB`, and then affecting your initial configuration as it was supposed to behave. Then in largers servers with quite more than 16 GiB RAM, there should be no limitation to your initial max readahead window size value.

Reply:
Clear.